### PR TITLE
Add text edit completion of `{% content_for "block" %}`

### PR DESCRIPTION
**Pull request by charlespwd**
## What are you adding in this PR?

- **Add snippet completion of `content_for 'block'`**
  - `{% content_for "block" %}` becomes `{% content_for "block", type: "$1", id: "$2" %}`
  - `{% content_for 'block' %}` becomes `{% content_for 'block', type: '$1', id: '$2' %}`
  - also works in `{% liquid %}` tag
  
- **Add the `applyEdits` custom matcher to make unit testing easier**

### normal expectations
```ts
// before
const doc = TextDocument.create(/* ... */);
expect(TextDocument.applyEdit(document, edit)).to.equal(expected);

```

```ts
// after
expect(edit).to.applyEdits(source, expected)
```

### asymmetric matching
```ts
// before
//                  char 12 ⌄   ⌄ char 16
const liquid = '{{ string | defa█ }}';

const textEdit: TextEdit = {
  newText: 'default',
  range: {
    end: { line: 0, character: 16 },
    start: { line: 0, character: 12 },
  },
};

await expect(provider).to.complete(liquid, [
  expect.objectContaining({
    label: 'default',
    insertTextFormat: InsertTextFormat.PlainText,
    textEdit,
  }),
]);

const textDocument = TextDocument.create('', 'liquid', 0, liquid.replace(CURSOR, ''));

expect(TextDocument.applyEdits(textDocument, [textEdit])).toBe('{{ string | default }}');
```

```ts
// after
await expect(provider).to.complete(liquid, [
  expect.objectContaining({
    label: 'default',
    insertTextFormat: InsertTextFormat.PlainText,
    textEdit: expect.applyEdits(liquid, '{{ string | default }}'),
  })
])
```

- **Add changesets**

## What did you learn?

- Vitest custom matchers can also be used as asymmetric matchers!

## Before you deploy

- [x] I included a minor bump `changeset`
- [x] My feature is backward compatible

- [x] I included a patch bump `changeset`

### CODE REVIEW

File: packages/theme-language-server-common/src/test/ApplyTextEditAssertion.spec.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  });
+
+  it('should also support asymetric matching', () => {
+    const source = 'hello world';
+    const doc = TextDocument.create('file:///file.liquid', 'liquid', 1, source);
+    const indexOfWorld = source.indexOf('world');
+    const textEdit = TextEdit.replace(
+      Range.create(doc.positionAt(indexOfWorld), doc.positionAt(indexOfWorld + 5)),
+      'planet',
+    );
```
Comment (aswamy) [inline packages/theme-language-server-common/src/test/ApplyTextEditAssertion.spec.ts:25]: you should stick this in a `before`

File: packages/theme-language-server-common/src/test/ApplyTextEditAssertion.spec.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    expect(textEdit).to.applyTextEdit(source, 'hello planet');
+    expect(textEdit).not.to.applyTextEdit('super world', 'hello planet');
+  });
+
+  it('should also support asymetric matching', () => {
```
Comment (aswamy) [inline packages/theme-language-server-common/src/test/ApplyTextEditAssertion.spec.ts:18]:
```suggestion
  it('should also support asymmetric matching', () => {
```

File: packages/theme-language-server-common/src/test/ApplyTextEditAssertion.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+interface CustomMatchers<R = unknown> {
+  applyTextEdit: (source: string, expected: string) => R;
+}
+
+declare module 'vitest' {
+  interface Assertion<T = any> extends CustomMatchers<TextEdit> {}
+  interface AsymmetricMatchersContaining extends CustomMatchers {}
+}
+
+export const applyTextEdit: RawMatcherFn = (
```
Comment (aswamy) [inline packages/theme-language-server-common/src/test/ApplyTextEditAssertion.ts:14]: ❤️

**Review by aswamy — APPROVED** [review]
amazing! tophat went great! thanks for making the helper method for checking text edits.

File: .changeset/nervous-islands-build.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/theme-language-server-common': patch
+---
+
+[Internal] Add `expect.applyTextEdit(source, expected)` custom matcher
```
Comment (jamesmengo) [inline .changeset/nervous-islands-build.md:5]: Does this do anything special?

**Pull request merged**